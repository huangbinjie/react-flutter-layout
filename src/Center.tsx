import React from 'react'
import Container from './Container'

export default function Center({ children, style, ...restProps }: React.HtmlHTMLAttributes<{}>) {
  return (
    <Container
      alignment="center"
      style={{ alignItems: 'center', ...style }}
      {...restProps}
    >
      {children}
    </Container>
  )
}
