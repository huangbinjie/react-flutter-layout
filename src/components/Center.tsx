import React from 'react'
import Container from './Container'
import { Style } from '../style'

export type CenterProps = {
  style: Style
  children: React.ReactNode
}

export default function Center({ children, style, ...restProps }: CenterProps) {
  return (
    <Container
      width="100%"
      height="100%"
      alignment="center"
      style={{ alignItems: 'center', ...style }}
      {...restProps}
    >
      {children}
    </Container>
  )
}
