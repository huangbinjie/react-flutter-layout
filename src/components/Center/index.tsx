import React, { CSSProperties } from 'react'
import { style as jss } from 'typestyle'
import classnames from 'classnames'

export type CenterProps = {
  className?: string
  style?: CSSProperties
  children?: React.ReactNode
}

export default function Center({ className = '', children, ...restProps }: CenterProps) {
  const classname = jss({
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: "100%",
    height: "100%"
  })

  return (
    <div
      debug-label="Center"
      className={classnames(classname, className)}
      {...restProps}
    >
      {children}
    </div>
  )
}
