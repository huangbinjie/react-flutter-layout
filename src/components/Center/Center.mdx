---
name: Center
menu: Components
---

import { Playground, Props } from 'docz'
import Container from '../Container'
import Center from '../Center'

## Center

提供居中效果的组件。

### Basic

<Playground>
  <Container width="100px" height="100px" color="#ececec">
    <Center>居中</Center>
  </Container>
</Playground>

## Props

<Props of={Center} />
