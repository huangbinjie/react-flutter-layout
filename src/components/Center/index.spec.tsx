import React from 'react'
import renderer from 'react-test-renderer'

import Center from '.'

describe('Center', () => {
  test('basic', () => {
    const tree = renderer.create(
      <Center>empty content</Center>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
