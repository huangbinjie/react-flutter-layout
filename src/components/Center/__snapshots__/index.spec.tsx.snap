// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center basic 1`] = `
<div
  className="fnlk30n"
  debug-label="Center"
>
  empty content
</div>
`;
